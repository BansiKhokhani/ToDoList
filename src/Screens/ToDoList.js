import React from 'react'
import { View, Text, StyleSheet, Image, TouchableOpacity } from 'react-native'
import Task from '../Components/Task'
import AddNewTaskButton from '../Components/AddNewTaskButton'


const ToDoList = () => {
  return (
    <View style={styles.mainWrapper}>
      {/* .............................header..................... */}
      <View style={styles.headerView}>
        <View style={styles.headerSubView}>
          <Text style={styles.headerViewText}>TODO APP</Text>
        </View>
      </View>
      {/* ..............MiddleView................... */}
      <View style={styles.middleView}>
        <Task />
        <Task />
        <AddNewTaskButton/>
      </View>
      {/* ...................FooterView..................... */}
      <View style={styles.footerView}>
        <View style={styles.footerSubView}>
          <View>
            <TouchableOpacity activeOpacity={1} >
              <Image source={require('../../assets/icon/Playlist.png')} style={styles.image} />
            </TouchableOpacity>
          </View>
          <View>
            <Text style={styles.footerSubViewText}>ALL</Text>
          </View>

        </View>
        <View style={styles.footerSubView}>
          <View>
            <TouchableOpacity activeOpacity={1} >
              <Image source={require('../../assets/icon/Tick.png')} style={styles.image} />
            </TouchableOpacity>
          </View>
          <View>
            <Text style={styles.footerSubViewText}>COMPLETED</Text>
          </View>

        </View>
      </View>
    </View>
  )
}

export default ToDoList

const styles = StyleSheet.create({
  mainWrapper: {
    flex: 1,
    backgroundColor: '#9395D3'
  },
  headerView:
  {
    flex: 0.12,
  },
  middleView: {
    flex: 0.8,
    backgroundColor: '#B3B7EE', paddingTop: '2%', paddingBottom: '2%'
  },
  footerView: {
    flex: 0.08,
    backgroundColor: '#ffff',
    flexDirection: 'row',
  },
  headerSubView: {
    flex: 1,
    justifyContent: 'flex-end',
    paddingBottom: '4%',
    paddingLeft: '4%',
  },
  headerViewText: {
    fontSize: 25,
    fontWeight: 'bold',
    color: '#ffff',

  },
  footerSubView: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  image: {
    width: 40, 
    height: 40,
    tintColor:'#9395D3'
  },
  footerSubViewText:{
    color:'#9395D3',fontWeight:'600'
  }
})