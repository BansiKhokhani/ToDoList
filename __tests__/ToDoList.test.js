import React from 'react';
import renderer from 'react-test-renderer';
import ToDoList from '../src/Screens/ToDoList';

test('renders correctly', () => {
  const tree = renderer.create(<ToDoList />).toJSON();
  expect(tree).toMatchSnapshot();
});