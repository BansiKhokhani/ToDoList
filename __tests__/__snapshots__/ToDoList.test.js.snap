// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#9395D3",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 0.12,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "flex-end",
          "paddingBottom": "4%",
          "paddingLeft": "4%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffff",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        TODO APP
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#B3B7EE",
        "flex": 0.8,
        "paddingBottom": "2%",
        "paddingTop": "2%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#ffff",
          "borderRadius": 15,
          "flex": 0.13,
          "flexDirection": "row",
          "margin": "2%",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingLeft": "4%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#9395D3",
              "fontSize": 17,
              "fontWeight": "bold",
            }
          }
        >
          TODO TILTE
        </Text>
        <Text
          style={
            {
              "color": "#bfc0f5",
              "fontWeight": "bold",
            }
          }
        >
          Due Date : 12/9/2024
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 0.5,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "paddingRight": "4%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/icon/Pencil.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/icon/Trash.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/icon/CheckCircle.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#ffff",
          "borderRadius": 15,
          "flex": 0.13,
          "flexDirection": "row",
          "margin": "2%",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingLeft": "4%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#9395D3",
              "fontSize": 17,
              "fontWeight": "bold",
            }
          }
        >
          TODO TILTE
        </Text>
        <Text
          style={
            {
              "color": "#bfc0f5",
              "fontWeight": "bold",
            }
          }
        >
          Due Date : 12/9/2024
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 0.5,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "paddingRight": "4%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/icon/Pencil.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/icon/Trash.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/icon/CheckCircle.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "bottom": 10,
          "position": "absolute",
          "right": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icon/AddNewButton.png",
            }
          }
          style={
            {
              "height": 70,
              "width": 70,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#ffff",
        "flex": 0.08,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/icon/Playlist.png",
              }
            }
            style={
              {
                "height": 40,
                "tintColor": "#9395D3",
                "width": 40,
              }
            }
          />
        </View>
      </View>
      <View>
        <Text
          style={
            {
              "color": "#9395D3",
              "fontWeight": "600",
            }
          }
        >
          ALL
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/icon/Tick.png",
              }
            }
            style={
              {
                "height": 40,
                "tintColor": "#9395D3",
                "width": 40,
              }
            }
          />
        </View>
      </View>
      <View>
        <Text
          style={
            {
              "color": "#9395D3",
              "fontWeight": "600",
            }
          }
        >
          COMPLETED
        </Text>
      </View>
    </View>
  </View>
</View>
`;
